mapie/subsample.py

```diff
@@ -66,12 +68,12 @@ def split(
 
         Yields
         ------
-        train : ArrayLike of shape (n_indices_training,)
+        train : NDArray of shape (n_indices_training,)
             The training set indices for that split.
-        test : ArrayLike of shape (n_indices_test,)
+        test : NDArray of shape (n_indices_test,)
             The testing set indices for that split.
         """
-        indices = np.arange(len(X))
+        indices = np.arange(_num_samples(X))
```

Comment:
why not keeping len ?

Reply:
Because not all `ArrayLike` objects have a `__len__` method, so mypy complain about it.